cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -479,6 +479,29 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		}
 	}
 
+	if a.ScaleUpToNodeGroupMinSizeEnabled {
```

Comment:
I wonder if it wouldn't make more sense to do this after scale down. Otherwise we may end up in a situation in which CA has empty nodes in one node group, waiting to be removed, but CA doesn't delete them, being busy with creating more empty nodes (that aren't actually going to help any pods) in another node group instead.

Reply:
Yes, I agreed, the scale down doesn't have to wait for the scale up and the cool down period. This also helps reduce the chance of hitting the cluster resource limit.